src/main/java/org/example/services/JobRoleService.java

```diff
@@ -19,10 +20,19 @@ public JobRoleService(final JobRoleDao jobRoleDao,
 
     public List<JobRoleResponse> getAllJobRoles() throws SQLException {
         try {
-            return jobRoleMapper.mapOrderListToJobRoleResponseList(
+            return jobRoleMapper.mapJobRoleListToJobRoleResponseList(
                     jobRoleDao.getJobRoles());
         } catch (SQLException e) {
             throw new SQLException(e);
         }
     }
+    public List<JobRoleDetailedResponse> getJobRoleById(final int id)
```

Comment:
This should return a single object... And also think about what happens if the user passes an id that doesnt exist... It wont produce a SqlException but you might want to think about returning a custom exception like DataNotFoundException or something similar